### PR TITLE
[RW-5434][risk=no] Update Puppeteer test for new Cohort Builder UI

### PR DESCRIPTION
**Pull request by dolbeew**
- Update existing tests for new cohort builder UI and workflow (converted modal to full page, use sidebar for selections, modifiers and attributes)
- Add new class for testing the `HelpSidebar` component
- Includes minor updates/cleanup of cohort builder components needed for tests

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] I have run and tested this change locally
- [x] I have run the E2E tests on ths change against my local UI + API server with `yarn test:local`

### CODE REVIEW

**Comment by dolbeew** [comment]
Not sure why the cohorts-create test is failing in circle-ci. It passes locally for me.

File: e2e/app/page/cohort-search-page.ts [hunk 1]
```diff
@@ -124,22 +120,32 @@ export default class CohortCriteriaModal extends Modal {
...
+    return Button.findByName(this.page, {normalizeSpace: LinkText.FinishAndReview}).then(button => button.click());
+  }
+
+  async viewAndSaveCriteria(): Promise<void> {
+    const finishAndReviewButton = await Button.findByName(this.page, {name: LinkText.FinishAndReview});
+    await finishAndReviewButton.waitUntilEnabled();
+    await finishAndReviewButton.click();
+
+    // Click Save Criteria button in sidebar
+    const helpSidebar = await new HelpSidebar(this.page);
```
Comment (aweng98) [inline e2e/app/page/cohort-search-page.ts:132]: nit: remove `await`

File: e2e/app/page/cohort-search-page.ts [hunk 2]
```diff
@@ -94,28 +108,10 @@ export default class CohortCriteriaModal extends Modal {
...
 
-    const selectMenu = await SelectMenu.findByName(this.page, {ancestorLevel: 2}, this);
-    await selectMenu.clickMenuItem(filterSign);
-
-    const numberField = await this.page.waitForXPath(`${this.xpath}//input[@type="number"]`, {visible: true});
-    await numberField.type(String(filterValue));
-
-    await this.clickButton(LinkText.Calculate);
-    const participantResult = await this.waitForParticipantResult();
+    const helpSidebar = await new HelpSidebar(this.page);
```
Comment (aweng98) [inline e2e/app/page/cohort-search-page.ts:111]: nit: remove `await` in class constructor call.

File: e2e/app/component/help-sidebar.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+import SelectMenu from 'app/component/select-menu';
+import Container from 'app/container';
+import Button from 'app/element/button';
+import {FilterSign} from 'app/page/cohort-search-page';
+import {LinkText} from 'app/text-labels';
+import {buildXPath} from 'app/xpath-builders';
+import {ElementType} from 'app/xpath-options';
+import {waitForNumericalString} from 'utils/waits-utils';
+import {waitWhileLoading} from '../../utils/test-utils';
```
Comment (aweng98) [inline e2e/app/component/help-sidebar.ts:11]: nit: remove `../..` before `utils`

File: e2e/app/component/help-sidebar.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    // Before add criteria, first check for nothing in Selected Criteria Content Box.
+    await this.page.waitForXPath(removeSelectedCriteriaIconSelector,{hidden: true});
+
+    await this.clickSidebarButton(LinkText.AddThis);
+    // After add criteria, should flip to selection list section
+    await this.waitUntilSectionVisible(SectionSelectors.SelectionList);
+    // Look for X (remove) icon for indication that add succeeded.
+    await this.page.waitForXPath(removeSelectedCriteriaIconSelector,{visible: true});
+
+    // dialog close after click FINISH button.
```
Comment (aweng98) [inline e2e/app/component/help-sidebar.ts:49]: old comment. no more modal window, right?

**Review by aweng98 — APPROVED** [review]
LGTM

File: e2e/app/component/help-sidebar.ts (hunk 3 above)
Comment (dolbeew) [reply e2e/app/component/help-sidebar.ts]: Done

File: e2e/app/component/help-sidebar.ts (hunk 4 above)
Comment (dolbeew) [reply e2e/app/component/help-sidebar.ts]: Nice catch. Fixed.

File: e2e/app/page/cohort-search-page.ts (hunk 2 above)
Comment (dolbeew) [reply e2e/app/page/cohort-search-page.ts]: Done.

File: e2e/app/page/cohort-search-page.ts (hunk 1 above)
Comment (dolbeew) [reply e2e/app/page/cohort-search-page.ts]: Done.

**Pull request merged**